tests/worker/file_util_test.py

```diff
@@ -65,3 +65,18 @@ def test_bz2_file(self):
 
     def test_gzip_bytestring(self):
         self.assertEqual(un_gzip_bytestring(gzip_bytestring(b'contents')), b'contents')
+
+    def test_tar_exclude_ignore(self):
+        dir = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'files/dir3')
+        temp_dir = tempfile.mkdtemp()
+        self.addCleanup(lambda: remove_path(temp_dir))
+        output_dir = os.path.join(temp_dir, 'output')
+
+        un_tar_directory(tar_gzip_directory(dir, ignore_file_name='.tarignore'), output_dir, 'gz')
+        output_dir_entries = os.listdir(output_dir)
+        self.assertIn('dir4', output_dir_entries)
```

Comment:
Could we name the files/directories in such a way to make it more obvious which ones are meant to be included versus excluded?

Reply:
Yes, that makes it much more clear.